### PR TITLE
Add new schema and use the placeholder user to query decks. Also updated the seed file accordingly

### PR DESCRIPTION
**Pull request by SlyPuffin**
User isn't used at this point, but hopefully this helps us start building users into the flow of various parts of the app.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -12,13 +12,24 @@ generator client {
...
 model Deck {
   id              String    @id @default(uuid())
   createdAt       DateTime  @default(now())
   updatedAt       DateTime  @updatedAt
   type            String    @db.VarChar(255)
   name            String    @db.Text
   cards           Card[]
+  userId          String
+  user            User      @relation(fields: [userId], references: [id], onDelete: Cascade)
+  @@index([userId])
```
Comment (SlyPuffin) [inline prisma/schema.prisma:32]: Added in the user and relation to decks. Nothing special at this point--it's the same relational format as cards to decks.

File: prisma/seed.ts [hunk 2]
```diff
@@ -3,71 +3,78 @@ import { PrismaClient } from "@prisma/client";
 const prisma = new PrismaClient();
 
 async function main() {
-  const deck = await prisma.deck.create({
+  const user = await prisma.user.create({
```
Comment (SlyPuffin) [inline prisma/seed.ts:6]: This diff is really ugly, but it's really just a change to add a user with a deck array which in turn has a card array. One more layer of confusion!

File: src/backend/router/index.ts [hunk 3]
```diff
@@ -5,9 +5,23 @@ import { prisma } from "@/backend/utils/prisma";
 
 export const appRouter = trpc
   .router()
-  .query("get-decks", {
+  .query("get-user", {
+    async resolve() {
+      const userFromDb = await prisma.user.findFirst();
```
Comment (SlyPuffin) [inline src/backend/router/index.ts:10]: Just fetching the first user--this is something I imagine we'll build on when Auth enters the picture.

File: src/backend/router/index.ts [hunk 4]
```diff
@@ -5,9 +5,23 @@ import { prisma } from "@/backend/utils/prisma";
...
+      return { user: userFromDb };
+    },
+  })
+  .query("get-decks-for-user", {
+    input: z.object({
+      id: z.string(),
```
Comment (SlyPuffin) [inline src/backend/router/index.ts:16]: Fetch the decks based on user id.

File: src/backend/router/index.ts [hunk 5]
```diff
@@ -41,25 +55,27 @@ export const appRouter = trpc
...
       return newCard;
-    }
+    },
   })
   .mutation("add-deck", {
     input: z.object({
       name: z.string(),
-      type: z.string()
+      type: z.string(),
+      userId: z.string(),
```
Comment (SlyPuffin) [inline src/backend/router/index.ts:67]: Add user id to adding new decks.

File: src/pages/index.tsx [hunk 6]
```diff
@@ -4,9 +4,18 @@ import Link from "next/link";
...
 
 export default function Home() {
-  const { data, isLoading } = trpc.useQuery(["get-decks"], {
+  const userQuery = trpc.useQuery(["get-user"], {
     refetchOnWindowFocus: false,
   });
+  const { data, isLoading } = trpc.useQuery(
+    [
+      "get-decks-for-user",
+      { id: userQuery.data?.user?.id ? userQuery.data?.user?.id : "" },
```
Comment (SlyPuffin) [inline src/pages/index.tsx:13]: Get the user, and if we have a user pulled in, we fetch the decks for that user.
This line (13) is really ugly. Again, this flow will likely change with auth, but stacking queries is something to look into more deeply.

File: src/backend/router/index.ts (hunk 5 above)
Comment (nleech16) [reply src/backend/router/index.ts:67]: Little bit of a conflict with my change, but that may or may not actually go in and/or will be easy to fix.

File: src/backend/router/index.ts [hunk 7]
```diff
@@ -31,7 +45,7 @@ export const appRouter = trpc
       back: z.string(),
       source: z.string(),
       xp: z.number(),
-      deckId: z.string()
+      deckId: z.string(),
```
Comment (nleech16) [inline src/backend/router/index.ts:48]: This just a typescript style thing? Seeing lots of these ending commas added

**Review by nleech16 — APPROVED** [review]

**Pull request merged**